### PR TITLE
Filter npm module defintions from catalogEntriesByType

### PR DESCRIPTION
**Pull request by ryanlabouve**
I created this PR to help explain the issue I reported here: https://github.com/ember-cli/ember-resolver/issues/161

I understand that this may not the the desired solution. However if this is useful, or if there's some other direction I can take to help solve this problem, I'm happy to do so. Just let me know :)

### CODE REVIEW

**Comment by nathanhammond** [comment]
This is not a viable fix, unfortunately. The current implementation pulls in everything from `require.entries` which has `${type}s/` anywhere in the path. It's likely that people are depending upon this behavior (whether or not that's a good thing). Different places I would look to fix this:

- Prevent data_adapter from eagerly looking up all models (which means that it will be available in a broken format, but never passed to `lookupFactory`.
- Have `ember-browserify` rewrite the module name at build time.
- ???

Unfortunately both of those are hard.

File: tests/unit/container-debug-adapter-test.js [hunk 1]
```diff
@@ -110,3 +110,18 @@ test("Pods podModulePrefix support", function(assert) {
   assert.equal(models[0], 'user', "the name is correct");
   assert.equal(models[1], 'users/user', "the name is correct");
 });
+
+test('Ignores browserify modules', function(assert) {
+  assert.expect(1);
+  def('npm:cool/model/package');
```
Comment (nathanhammond) [inline tests/unit/container-debug-adapter-test.js]: This test will never fail. `model` needs to be pluralized.

File: tests/unit/container-debug-adapter-test.js [hunk 2]
```diff
@@ -110,3 +110,18 @@ test("Pods podModulePrefix support", function(assert) {
...
+  var models = containerDebugAdapter.catalogEntriesByType('model');
+  assert.equal(models.length, 0, 'Ignores the browserify modules');
+});
+
+test('Ignores modules that are not part of the main app', function(assert) {
+  assert.expect(1);
+  def('liquid/awesome/model');
```
Comment (nathanhammond) [inline tests/unit/container-debug-adapter-test.js]: Pluralize, add trailing slash.

**Comment by igorT** [comment]
@nathanhammond do you have an example for where this fix fails? We originally filtered out things starting with `npm` but @rwjblue suggested we should filter out other spurious matches. `dataAdapter` is definitely the wrong place to fix this, because the method called `catalogEntriesByType` shouldn't be returning modules which don't match the type.

**Comment by nathanhammond** [comment]
@igorT [Given this](https://github.com/ember-cli/ember-resolver/blob/d8e2b892ab562d2689a7ffbe6c6f59d85da05d89/addon/container-debug-adapter.js#L89) if you have an addon which defines a model which ended up at `some-addon/models/foo` then this change would break it as the `foo` model would have previously been available.

Forcing everything to be in the application namespace is too heavy-handed.

**Comment by rwjblue** [comment]
> Forcing everything to be in the application namespace is too heavy-handed.

FWIW, this is untrue.  Models that are not in the applications namespace **can never be resolved** anyways (since we only resolve things in the apps namespace).

File: tests/unit/container-debug-adapter-test.js (hunk 2 above)
Comment (rwjblue) [inline tests/unit/container-debug-adapter-test.js]: @nathanhammond - This particular module stub is emulating "pod models" (your favorite 😉 ), and singular form is correct here.

File: tests/unit/container-debug-adapter-test.js (hunk 1 above)
Comment (rwjblue) [inline tests/unit/container-debug-adapter-test.js]: Confirm on this one. Should be `models` here...

**Comment by ryanlabouve** [comment]
Sure thing! Sorry for delay. Will do it tonight.

**Comment by rwjblue** [comment]
Yep, this is definitely going the right direction.  I think I only had a couple of minor changes needed to land it...

**Comment by stefanpenner** [comment]
This code actually now resides within ember itself: https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-extension-support/lib/container_debug_adapter.js (since ember 1.7, but this library accidentally overrode it)

**Pull request closed without merging**